### ISSUE TITLE
[bug] false-positive for `@eslint-react/no-unnecessary-use-memo` with unused variable

### ISSUE THREAD

**Issue by silverwind** [Rel1cx/eslint-react#1404]
### Describe the bug

```tsx
function Component() {
  const foo = "foo";
  const bar = useMemo(() => foo, [foo]);
}
```

This incorrectly triggers this error at the `useMemo`:

```
bar is only used inside 1 useEffect, which may be unnecessary. You can move the computation into useEffect directly and merge the dependency arrays   @eslint-react/no-unnecessary-use-memo
```

`bar` is used in 0 `useEffect`, not 1 as stated in the message. It's an edge case because actual code would make use of `bar` which makes the error go away, but I find it distracting having this error show up temporarily while writing the code.

### Platform and versions

```Text
Node.js v24.12.0, @eslint-react v2.6.4
```
